src/Core/src/Handlers/Button/ButtonHandler.iOS.cs

```diff
@@ -182,16 +183,27 @@ public void Disconnect(UIButton platformView)
 				platformView.TouchUpInside -= OnButtonTouchUpInside;
 				platformView.TouchUpOutside -= OnButtonTouchUpOutside;
 				platformView.TouchDown -= OnButtonTouchDown;
+				platformView.TouchCancel -= OnButtonTouchCancel;
 			}
 
-			void OnButtonTouchUpInside(object? sender, EventArgs e)
-			{
-				if (VirtualView is IButton virtualView)
-				{
-					virtualView.Released();
-					virtualView.Clicked();
-				}
-			}
+			void HandleButtonInteraction()
+            {
+                if (VirtualView is IButton virtualView)
```

Comment:
`VirtualView` is an Button, can directly do:

```
VirtualView?.Released();
VirtualView?.Clicked();
```

Reply:
I have added the changes as per your suggestion. Could you please validate it once from your end?